src/main/webapp/app/exercises/shared/result/result.component.ts

```diff
@@ -78,6 +79,7 @@ export class ResultComponent implements OnInit, OnChanges {
         private http: HttpClient,
         private modalService: NgbModal,
         private exerciseService: ExerciseService,
+        @Optional() private exerciseCacheService: ExerciseCacheService,
```

Comment:
Why is this optional?

Reply:
We do not always want to cache. The cache service is provided on a component basis and not in the root. i.e. if you navigate out of a cached component the cache is not persistent